app/controllers/concerns/locale.rb

```diff
@@ -0,0 +1,51 @@
+module Concerns::Locale
+  extend ActiveSupport::Concern
+
+  included do
+    before_filter :set_locale
+  end
+
+  def explicitly_requested_language
+    params[:locale]
+  end
+
+  def user_settings_language
+    current_user.locale if current_user
+  end
+
+  def session_language
+    session[:locale]
+  end
+
+  def browser_language
+    request.env['HTTP_ACCEPT_LANGUAGE'] ? request.env['HTTP_ACCEPT_LANGUAGE'].scan(/^[a-z]{2}/).first : nil
```

Comment:
is there no better way or at least a gem for this?

Reply:
Interesting, I'll open an issue for that. --> #574